compiler/q-implant/src/QImplant.cpp

```diff
@@ -251,6 +271,50 @@ void QImplant::write(loco::Graph *g)
     }
   }
 
+  std::set<luci::CircleNode *> visit;
+  std::deque<luci::CircleNode *> que;
+  for (auto input_node : loco::input_nodes(g))
+  {
+    auto node = loco::must_cast<luci::CircleNode *>(input_node);
+    que.emplace_back(node);
+  }
+
+  // check whether node can share its successor node quantization parameter
+  while (!que.empty())
+  {
+    auto node = que.front();
+    que.pop_front();
+
+    // skip when node is output
+    if (node->opcode() == luci::CircleOpcode::CIRCLEOUTPUT)
+    {
+      continue;
+    }
+
+    auto quantparam = node->quantparam();
+    THROW_UNLESS(quantparam);
```

Comment:
`node` can be a non-quantizable type (ex: int32 or bool). `continue` would be enough.

Exception has to be used carefully, because users will instantly experience the failure. So, we typically throw an exception when the input of program (circle model and param.json in this case) is not valid, or users misuse the tool.

Reply:
I intend to make `exception` early when `exception` occurs later.
I never thought about non-quantizable type before, and just continuing would works fine!

So just `continue` would be better!